packages/matrix-client-server/src/index.test.ts

```diff
@@ -654,6 +754,19 @@ describe('Use configuration file', () => {
         expect(response.body).toHaveProperty('error')
         expect(response.body).toHaveProperty('errcode', 'M_USER_IN_USE')
       })
+      it('should refuse a request without User Agent', async () => {
```

Comment:
Why did you require a User-Agent in this particular API ?

Reply:
Since the user_agent field in Synapse's matrixDB has the constraint "NOT NULL", but the spec doesn't say it Is required, so it might not be necessary to include it. I included this constraint to correctly fill the DB but this might change in the future

Reply:
If it's not in spec, maybe we can populate User-Agent with "undefined" when missing instead of rejecting request

Reply:
Alright, I'll change that